os/board/rtl8730e/src/rtl8730e_st7701.c

```diff
@@ -0,0 +1,202 @@
+/****************************************************************************
+ *
+ * Copyright 2024 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/lcd/st7701.h>
+#include <tinyara/mipidsi/mipi_dsi.h>
+#include "objects.h"
+#include "PinNames.h"
+#include "gpio_api.h"
+
+
+static int rtl8730_st7701_lcd_init(void);
+static int rtl8730_st7701_gpio_reset(void);
+static int rtl8730_st7701_cache_invalidate(u32 *buffer, int size);
+static int rtl8730_st7701_lcd_reload(void);
+static int rtl8730_st7701_lcd_enable(void);
+static int rtl8730_st7701_lcd_layer_enable(int layer, bool enable);
+static void lcd_put_area(u8 *LcdImgBuffer, u32 x1, u32 y1, u32 x2, u32 y2);
+static void LcdcEnable(void);
+
+struct st7701_config_s g_rtl8730_st7701_config_s = {
+	.init = rtl8730_st7701_lcd_init,
+	.reset = rtl8730_st7701_gpio_reset,
+    .cache_invalidate = rtl8730_st7701_cache_invalidate,
+    .lcd_reload = rtl8730_st7701_lcd_reload,
+	.lcd_enable = rtl8730_st7701_lcd_enable,
+	.lcd_layer_enable = rtl8730_st7701_lcd_layer_enable,
+	.lcd_put_area = lcd_put_area,
+};
+
+typedef struct {
+	u8 blue;
+	u8 green;
+	u8 red;
+	u8 alpha;
+} argb8888_t;
+
+typedef struct {
+	u8 blue;
+	u8 green;
+	u8 red;
+} rgb888_t;
```

Comment:
They look common structures, not st7701 specific. How about moving them to other common file? Because other file also uses this and defines there duplicately.